tgui/packages/tgui/interfaces/Helm.tsx

```diff
@@ -0,0 +1,352 @@
+                    onClick={() => act('roll', { roll: 4 })}
+                  />
+                </Table.Cell>
+              </Table.Row>
+            </Table>
+          </Table.Cell>
+        </Table.Row>
+      </Table>
+      <Button
+        content={
+          'Manual Control ' + (data.manual_control ? 'Engaged' : 'Disengaged')
+        }
+        icon="cog"
+        iconSpin={data.manual_control ? '0.0' : null}
+        onClick={() => act('manual', { manual: true })}
+      />
+    </Section>
+  );
+};
+
+const AutopilotSection = function (act, data) {
+  return (
+    <Section title="Autopilot">
+      <Table>
+        <Table.Row>
+          <Table.Cell>Target:</Table.Cell>
+          <Table.Cell>
+            {data.dest ? (
+              <>
+                <Button
+                  content={data.d_x}
+                  onClick={() => act('setx', { setx: true })}
+                />
+                <Button
+                  content={data.d_y}
+                  onClick={() => act('sety', { sety: true })}
+                />
+              </>
+            ) : (
+              <Button
+                content="None"
+                onClick={() => {
+                  act('setx', { setx: true });
+                  act('sety', { sety: true });
+                }}
+              />
+            )}
+          </Table.Cell>
+        </Table.Row>
+        <Table.Row>
+          <Table.Cell>Speed limit:</Table.Cell>
+          <Table.Cell>
+            <Button
+              content={data.speedlimit + ' Gm/h'}
+              onClick={() => act('speedlimit', { speedlimit: true })}
+            />
+          </Table.Cell>
+        </Table.Row>
+        <Table.Row>
+          <Button
+            content={'Autopilot ' + (data.autopilot ? 'Engaged' : 'Disengaged')}
+            icon="cog"
+            iconSpin={data.autopilot ? '0.0' : null}
+            disabled={!data.dest}
+            onClick={() => act('apilot', { apilot: true })}
+          />
+        </Table.Row>
+      </Table>
+    </Section>
+  );
+};
+
+const NavSection = function (act, data) {
+  return (
+    <Section title="Navigation Data">
+      <Table>
+        <Table.Row>
+          <Table.Cell>Location:</Table.Cell>
+          <Table.Cell>{data.sector}</Table.Cell>
+        </Table.Row>
+        <Table.Row>
+          <Table.Cell>Coordinates:</Table.Cell>
+          <Table.Cell>
+            {data.s_x} : {data.s_y}
+          </Table.Cell>
+        </Table.Row>
+        <Table.Row>
+          <Table.Cell>Scan data:</Table.Cell>
+          <Table.Cell>{data.sector_info}</Table.Cell>
+        </Table.Row>
+        <Table.Row>
+          <Table.Cell>Status:</Table.Cell>
+          <Table.Cell>{data.landed}</Table.Cell>
+        </Table.Row>{' '}
+      </Table>
+    </Section>
+  );
+};
+
+const PosSection = function (act, data) {
+  return (
+    <Section title="Saved Positions">
+      <Table>
+        {data.locations.length ? (
```

Comment:
add `data.locations &&` before the length check

You need to check for if variable exists first, otherwise length will error

Reply:
addressed